### DISCUSSION TITLE
Programmation  of a gopro hero 11

### DISCUSSION THREAD

**Discussion by DavidBarriere** [gopro/labs#638] [General]
Dear developpers of Gopro Labs,
I'd like to use my gopro camera to record during 15min each hour and send movies and metafiles to my personal cloud.
Additionnaly i'd like that the camera detect night/day conditions and and automattically switch between both mode.

It sounds simple with Gopro Labs using loops "start" and "stop" commands and both settings and variables commands.
But I'am looking for examples of codes to understand how to program the sequence and how generate the QRCode to set my camera.

Thank you.

David

**Comment by dnewman-gpsw** [comment]
This is doable but complex.  Start smaller. The trickiest part will be the cloud sync, as scripting that from Labs can be unreliable. 

The visual scripting is so new, there isn't a good collection of examples. https://gopro.github.io/labs/build

Here are few I've created:
USB photo trigger
![image](https://github.com/gopro/labs/assets/10856084/06480f91-2cb1-459d-b2cf-3aa6b8d11615)

Dash Cam  
![image](https://github.com/gopro/labs/assets/10856084/fb9ba024-06ae-4c9d-8c93-b5671a26c7bb)

Dive Mode
![image](https://github.com/gopro/labs/assets/10856084/1a29f76e-9ce9-4982-83ca-2de062ce3d82)

Photo Booth (use this last week, for a motion trigger photo booth camera)
![image](https://github.com/gopro/labs/assets/10856084/925f29bc-c289-44a5-b25e-07c62c9c6322)

**Reply by DavidBarriere** [reply]
Thank you for your rapid reply.
I saw the visual scripting tool and I found it useful but I am still desoriented as I am a python speaker :)
As an example I am confused with the loop using.

How it's work to do such a simple stuff:

for i in range(0,23):
    if i < 9:
        startingTime = str( '0' + i + ':00' )
        endTime = str( '0' + i + ':15' )
    else:
        startingTime = str( i + ':00' )
        endTime = str( i + ':15 ') 
startCamera( startingTime )
endCamera( endTime )
generateQRCode()

**Comment by DavidBarriere** [comment]
something like that and I repeat the block by changing starting and ending times ?
![Screenshot 2023-11-08 at 14-18-50 Go pro blockly](https://github.com/gopro/labs/assets/101575749/4c577f8c-eb43-40a7-930c-9c1bb56cd8fe)

**Comment by dnewman-gpsw** [comment]
Do something like this:
![15min per hour-11_8_2023, 09_21_00](https://github.com/gopro/labs/assets/10856084/12480f38-1e45-4c7b-8403-a13193d7365f)
You can load this PNG into "Import Image" to continue your edit.

![image](https://github.com/gopro/labs/assets/10856084/109d3bac-c699-4adc-8f02-0ce4c2793a51)
